src/main/java/com/github/checkstyle/regression/Main.java

```diff
@@ -34,6 +74,160 @@ private Main() {
+    private static Options createOptions() {
+        final Options options = new Options();
+
+        final Option repo = Option.builder("r")
+                .longOpt(OPT_LOCAL_GIT_REPO)
+                .required()
+                .hasArg()
+                .desc("the path of the checkstyle repository")
+                .build();
+        repo.setRequired(true);
+        options.addOption(repo);
+
+        final Option branch = Option.builder("p")
+                .longOpt(OPT_PATCH_BRANCH)
+                .required()
+                .hasArg()
+                .desc("the name of the PR branch")
+                .build();
+        options.addOption(branch);
+
+        final Option tester = Option.builder("t")
+                .longOpt(OPT_CHECKSTYLE_TESTER)
+                .required(false)
+                .hasArg()
+                .desc("the path of the checkstyle-tester directory")
+                .build();
+        options.addOption(tester);
+
+        final Option stopAfterGenerateConfig = Option.builder()
+                .longOpt(OPT_STOP_AFTER_GENERATE_CONFIG)
+                .required(false)
+                .hasArg(false)
+                .desc("indicates that regression tool would stop after generating config")
+                .build();
+        options.addOption(stopAfterGenerateConfig);
+
+        return options;
+    }
+
+    /**
+     * Validates the parsed CLI arguments.
+     * @param args the parsed CLI arguments.
+     * @throws IllegalArgumentException the arguments are invalid
+     */
+    private static void validateArguments(Arguments args) {
+        if (!existAndIsDirectory(args.repoPath())) {
+            throw new IllegalArgumentException(
+                    "path of local git repo must exist and be a directory");
+        }
+        if (!args.stopAfterGenerateConfig() && !existAndIsDirectory(args.testerPath().get())) {
+            throw new IllegalArgumentException(
+                    "path of checkstyle tester must exist and be a directory");
+        }
+    }
+
+    /**
+     * Runs the regression tool.
+     * @param args the parsed CLI arguments.
+     * @throws Exception excute failure
+     */
+    private static void runRegression(Arguments args) throws Exception {
+        final File config = generateConfig(args.repoPath(), args.branch());
+        System.out.println("config generated at " + config.getAbsolutePath());
+        if (!args.stopAfterGenerateConfig()) {
+            final File report = ReportGenerator.generate(
+                    args.testerPath().get(), args.repoPath(), args.branch(), config);
+            System.out.println("report generated at " + report.getAbsolutePath());
+        }
+    }
+
+    /**
+     * Checks whether the file in given path exists and is a directory.
+     * @param path the path to check
+     * @return true if file exists and is a directory
+     */
+    private static boolean existAndIsDirectory(String path) {
+        final File file = new File(path);
```

Comment:
if `path` can be `null` or empty string, won't this incorrectly return that it exists?
verify first that `null` or empty string can get into here.